zipkin-collector/core/src/main/java/zipkin2/collector/Collector.java

```diff
@@ -106,16 +108,18 @@ public Collector build() {
   }
 
   public void accept(List<Span> spans, Callback<Void> callback) {
-    accept(spans, callback, Runnable::run);
+    accept(spans, callback, Runnable::run, true);
   }
 
   /**
-   * @param executor the executor used to enqueue the storage request.
+   *  Based on the async indicator collect spans synchronously or asynchronously
    *
-   * <p>Calls to get the storage component could be blocking. This ensures requests that block
-   * callers (such as http or gRPC) do not add additional load during such events.
+   * @param spans spans to store
+   * @param callback callback function to invoke on success or error
+   * @param executor the executor used to enqueue the storage request.
+   * @param async indicator used to determine whether to collect spans async or sync
    */
-  public void accept(List<Span> spans, Callback<Void> callback, Executor executor) {
+  public void accept(List<Span> spans, Callback<Void> callback, Executor executor, boolean async) {
```

Comment:
this is still a low level change. please lets discuss on chat? https://gitter.im/openzipkin/zipkin